<a href="https://colab.research.google.com/github/doruktopcu/GlobalAI-Hub-Python-Bootcamp-2022/blob/main/GlobalAiHubProject2_MoviesAnalysis.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

## 0) Libraries & Data 

In [ ]:
import pandas as pd
import matplotlib.pyplot as plt
import seaborn as sns

In [ ]:
#First we uploaded our dataset to colab.
#We will save it as a DataFrame with pandas.
movies_df = pd.read_csv('NetflixOriginals.csv', sep=',', encoding='latin-1')

In [ ]:
#movies_df is ready to analyse.
movies_df

Title               Genre    Premiere  Runtime  IMDB Score  \
0               1922  Horror/Crime drama  20/10/2017      102         6.3   
1              44764               Drama  10/10/2018      144         6.8   
2              44788        Comedy-drama  29/03/2019      124         5.8   
3       #REALITYHIGH              Comedy  09/08/2017       99         5.2   
4               13th         Documentary  10/07/2016      100         8.2   
..               ...                 ...         ...      ...         ...   
579             XOXO               Drama  26/08/2016       92         5.3   
580       Yeh Ballet               Drama  21/02/2020      117         7.6   
581          Yes Day              Comedy  03/12/2021       86         5.7   
582  You've Got This     Romantic comedy  10/02/2020      111         5.8   
583            Zion          Documentary  08/10/2018       11         7.2   

    Language  
0    English  
1    English  
2    Marathi  
3    English  
4    English  
..       ...  
579  English  
580    Hindi  
581  English  
582  Spanish  
583  English  

[584 rows x 6 columns]

In [ ]:
movies_df.isnull().sum()

Title         0
Genre         0
Premiere      0
Runtime       0
IMDB Score    0
Language      0
dtype: int64

No null values in the dataset.

In [ ]:
movies_df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 584 entries, 0 to 583
Data columns (total 6 columns):
 #   Column      Non-Null Count  Dtype  
---  ------      --------------  -----  
 0   Title       584 non-null    object 
 1   Genre       584 non-null    object 
 2   Premiere    584 non-null    object 
 3   Runtime     584 non-null    int64  
 4   IMDB Score  584 non-null    float64
 5   Language    584 non-null    object 
dtypes: float64(1), int64(1), object(4)
memory usage: 27.5+ KB


In [ ]:
#Converting Premiere column to date type.
movies_df['Premiere'] = pd.to_datetime(movies_df.Premiere)

In [ ]:
movies_df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 584 entries, 0 to 583
Data columns (total 6 columns):
 #   Column      Non-Null Count  Dtype         
---  ------      --------------  -----         
 0   Title       584 non-null    object        
 1   Genre       584 non-null    object        
 2   Premiere    584 non-null    datetime64[ns]
 3   Runtime     584 non-null    int64         
 4   IMDB Score  584 non-null    float64       
 5   Language    584 non-null    object        
dtypes: datetime64[ns](1), float64(1), int64(1), object(3)
memory usage: 27.5+ KB


Premiere column is now a datetime64[ns] type

In [ ]:
movies_df['Year'] = movies_df['Premiere'].dt.year

In [ ]:
movies_df.head()

Title               Genre   Premiere  Runtime  IMDB Score Language  \
0          1922  Horror/Crime drama 2017-10-20      102         6.3  English   
1         44764               Drama 2018-10-10      144         6.8  English   
2         44788        Comedy-drama 2019-03-29      124         5.8  Marathi   
3  #REALITYHIGH              Comedy 2017-09-08       99         5.2  English   
4          13th         Documentary 2016-10-07      100         8.2  English   

   Year  
0  2017  
1  2018  
2  2019  
3  2017  
4  2016

We now have a year section which will make our job easier in answering some questions about the data set.

## 1) Genre Analysis

In [ ]:
#We will find how many genres in the dataset.
genre_list_df = pd.DataFrame(movies_df.groupby('Genre').Genre.count())

In [ ]:
genre_list_df

Genre
Genre                  
Action                7
Action comedy         5
Action thriller       1
Action-adventure      1
Action-thriller       3
...                 ...
War                   2
War drama             2
War-Comedy            1
Western               3
Zombie/Heist          1

[115 rows x 1 columns]

In genre_list_df we can see how many genres there are (115) and how many movies that belong to that genre.

In [ ]:
genre_list_df = genre_list_df.rename(columns = {'Genre' : 'Count' })

In [ ]:
genre_list_df = genre_list_df.sort_values(by = 'Count', ascending = False)

In [ ]:
genre_list_df.head(10)

Count
Genre                 
Documentary        159
Drama               77
Comedy              49
Romantic comedy     39
Thriller            33
Comedy-drama        14
Crime drama         11
Biopic               9
Horror               9
Action               7

In this dataset most movie produced top 10 genre can be seen. Documentary genre dominated the list.

In [ ]:
genre_list_df.tail(10)

Count
Genre                            
Comedy horror                   1
Drama-Comedy                    1
Drama / Short                   1
Dance comedy                    1
Crime thriller                  1
Coming-of-age comedy-drama      1
Comedy/Horror                   1
Comedy/Fantasy/Family           1
Comedy mystery                  1
Zombie/Heist                    1

The last 10 genres can be seen in this table. But a question arises; can these genres be merged into a more generalised genre? Some of these genres seem utterly specific.

## 2) Language Analysis

In [ ]:
movies_language_df = pd.DataFrame(movies_df.groupby('Language').Language.count())

In [ ]:
movies_language_df = movies_language_df.rename(columns = {'Language' : 'Count' })

In [ ]:
#We will sort the the list in descending order.
movies_language_df = movies_language_df.sort_values(by = 'Count', ascending = False) 

In [ ]:
#Top 10 most used languages in movies in this dataset.
movies_language_df.head(10)

Count
Language         
English       401
Hindi          33
Spanish        31
French         20
Italian        14
Portuguese     12
Indonesian      9
Korean          6
Japanese        6
German          5

It is clear to say Hollywood is dominating the industry.

In [ ]:
#We can create a piechart to see which language has more percentage.

## 3) Runtime Analysis

In [ ]:
#We select the movies that has more runtime than 120 minutes and save it as a dataframe.
long_runtime_movies = movies_df[(movies_df.Runtime > 120)]

In [ ]:
long_runtime_movies.describe()


Runtime  IMDB Score
count   68.000000   68.000000
mean   133.382353    6.592647
std     13.393058    0.997507
min    121.000000    3.500000
25%    124.000000    6.075000
50%    131.000000    6.700000
75%    139.000000    7.200000
max    209.000000    8.500000

## 4) IMDB Score Anaylsis

In [ ]:
movies_df['IMDB Score'].describe()

count    584.000000
mean       6.271747
std        0.979256
min        2.500000
25%        5.700000
50%        6.350000
75%        7.000000
max        9.000000
Name: IMDB Score, dtype: float64

In [ ]:
#Top 10 by IMDB scores.
top_ten_by_imdb = movies_df.sort_values(by = 'IMDB Score', ascending = False).head(10)

In [ ]:
top_ten_by_imdb

Title         Genre   Premiere  \
121     David Attenborough: A Life on Our Planet   Documentary 2020-10-04   
145    Emicida: AmarElo - It's All For Yesterday   Documentary 2020-12-08   
412                      Springsteen on Broadway  One-man show 2018-12-16   
67    Ben Platt: Live from Radio City Music Hall  Concert Film 2020-05-20   
577  Winter on Fire: Ukraine's Fight for Freedom   Documentary 2015-10-09   
427        Taylor Swift: Reputation Stadium Tour  Concert Film 2018-12-31   
114                      Cuba and the Cameraman    Documentary 2017-11-24   
118                       Dancing with the Birds   Documentary 2019-10-23   
523        The Three Deaths of Marisela Escobedo   Documentary 2020-10-14   
383                                   Seaspiracy   Documentary 2021-03-24   

     Runtime  IMDB Score                  Language  Year  
121       83         9.0                   English  2020  
145       89         8.6                Portuguese  2020  
412      153         8.5                   English  2018  
67        85         8.4                   English  2020  
577       91         8.4  English/Ukranian/Russian  2015  
427      125         8.4                   English  2018  
114      114         8.3                   English  2017  
118       51         8.3                   English  2019  
523      109         8.2                   Spanish  2020  
383       89         8.2                   English  2021

In [ ]:
#IMDB scores of Documentaries that has premiered between 2019 January to 2020 June.


In [ ]:
#Top 10 movies by IMDB in different genres.
top_movies_by_genre = pd.DataFrame(movies_df.groupby(['Genre'])['IMDB Score'].max().nlargest(10))

In [ ]:
top_movies_by_genre

IMDB Score
Genre                                           
Documentary                                  9.0
One-man show                                 8.5
Concert Film                                 8.4
Animation/Christmas/Comedy/Adventure         8.2
Drama                                        7.9
Animation / Short                            7.8
Crime drama                                  7.8
Making-of                                    7.7
Musical / Short                              7.7
War drama                                    7.7

In [ ]:
#Movies that has the highest IMDB scores in English. 
top_movies_by_genre = pd.DataFrame(movies_df.filter(like = 'English'))


## 5) Correlation Between Rt and IMDB

In [ ]:
#Correlation between runtime and IMDB score.
movies_df.corr()

Runtime  IMDB Score
Runtime     1.000000   -0.040896
IMDB Score -0.040896    1.000000

## 6) Year analysis

In [ ]:
year_movies_df = pd.DataFrame(movies_df.Year.value_counts())

In [ ]:
year_movies_df.rename(columns = {'Year' : 'Count'}, inplace = True)

In [ ]:
year_movies_df

Count
2020    183
2019    125
2018     99
2021     71
2017     66
2016     30
2015      9
2014      1

## TASKS

In [ ]:
#Avg. runtime of Hindi movies.

In [ ]:
#How many categories does Genre have.

In [ ]:
#Most number of movies released by years.

In [ ]:
#Movies in which languages has the lowest IMDB scores.

In [ ]:
#Which year has to most runtime sum.

In [ ]:
#Are there outliers in the data set?

In [ ]:
#Which languages are mostly used by genre?